Mc incident report

portermahoney opened the pull request:
Add an incident report form and list.  Similar to the material request we will have to discuss the workflow with them.  Also switched to viewsets over generics which is less code.

aaronk commented:
Generally looks great.

Is the ifxuser on the IncidentReport the author?  Is there a reason this should be nullable?

portermahoney commented:
fixed the field integrity and server side validation of that, stole showErrorMessage function from pubs.  We can use vuelidate for client side validation if we want, it changed in a recent release to not show the * for required, instead I am just using css and did not set up client side validation at this time.  Deleted the catch all from django so that non api routes show a 404 in vue rather than a 200 with the index page as response.
